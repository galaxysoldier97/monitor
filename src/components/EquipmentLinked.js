import React from 'react';
import { Grid, Tooltip } from '@material-ui/core';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { t } from 'mt-react-library/functions';
import WarningIcon from '@material-ui/icons/Warning';
import { routePaths, ROUTES } from '../config/routes';

const EquipmentLinked = (props) => {
  const {eqmId, eqmCategory} = props;

  const getEquipmentLink = () => {
    switch (eqmCategory) {
      case 'CPE':
        return ROUTES.cpeInfo.path.replace(':equipmentId', eqmId);
      case 'ANCILLARY':
        return ROUTES.ancillaryEquipmentsInfo.path.replace(':equipmentId', eqmId);
      default:
        return routePaths.equipmentInfo.replace(':equipmentId', eqmId);
    }
  };

  if (eqmId && eqmCategory) {
    return <Link to={getEquipmentLink()}>{eqmId} <strong>-</strong> {eqmCategory}</Link>;
  }

  if (eqmId) {
    return (
      <Grid container direction="row" alignItems="center" spacing={1}>
        <Grid item>
          {eqmId}
        </Grid>
        <Grid item>
          <Tooltip title={t('equipmentCategoryUndefined')}><WarningIcon fontSize="small" style={{ color: 'orange' }} /></Tooltip>
        </Grid>
      </Grid>
    );
  }

  return <div>-</div>;
};

EquipmentLinked.propTypes = {
  eqmId: PropTypes.number || null,
  eqmCategory: PropTypes.string || null,
};

export default EquipmentLinked;
