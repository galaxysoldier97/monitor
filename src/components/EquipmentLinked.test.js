import React from 'react';
import renderer from 'react-test-renderer';
import EquipmentLinked from './EquipmentLinked';
import { BrowserRouter } from 'react-router-dom';

test('Equipment linked id and category null', () => {
  const component = renderer.create(<BrowserRouter>
    <EquipmentLinked pairedEqPath="test" eqmId={null} eqmCategory={null} /></BrowserRouter>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Equipment linked id not null and category null', () => {
  const component = renderer.create(<BrowserRouter>
    <EquipmentLinked pairedEqPath="test" eqmId={123} eqmCategory={null} /></BrowserRouter>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Equipment linked id null and category not null', () => {
  const component = renderer.create(<BrowserRouter>
    <EquipmentLinked pairedEqPath="test" eqmId={null} eqmCategory="test" /></BrowserRouter>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Equipment linked id and category not null', () => {
  const component = renderer.create(<BrowserRouter>
    <EquipmentLinked pairedEqPath="test" eqmId={123} eqmCategory="test" /></BrowserRouter>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
