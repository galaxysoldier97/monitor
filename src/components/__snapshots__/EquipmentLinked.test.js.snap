// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Equipment linked id and category not null 1`] = `
<a
  href="/equipment/123"
  onClick={[Function]}
>
  123
   
  <strong>
    -
  </strong>
   
  test
</a>
`;

exports[`Equipment linked id and category null 1`] = `
<div>
  -
</div>
`;

exports[`Equipment linked id not null and category null 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    123
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <svg
      aria-describedby={null}
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
      focusable="false"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "color": "orange",
        }
      }
      title={null}
      viewBox="0 0 24 24"
    >
      <path
        d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
      />
    </svg>
  </div>
</div>
`;

exports[`Equipment linked id null and category not null 1`] = `
<div>
  -
</div>
`;
